File: 02-tech/03-software-technologies/08-exam-preparation/01-cat-shop/js/models/cat.js
const Sequelize = require('sequelize');

/**
 *
 * @param {Sequelize.Sequelize} sequelize
 * @returns { Sequelize.Model }
 */
module.exports = function (sequelize)
{
    let Cat = sequelize.define("Cat",
    {
        name:
        {
            type: Sequelize.STRING,
            allowNull: false,
            // @ts-ignore
            required: true

        },
        nickname:
        {
            type: Sequelize.STRING,
            allowNull: false,
            required: true
        },
        price:
        {
            type: Sequelize.DOUBLE,
            allowNull: false,
            required: true
        }
    },
    {
        timestamps: false
    });

    return Cat;
};

File: 02-tech/03-software-technologies/08-exam-preparation/01-cat-shop/csharp/CatShop/Models/Cat.cs
namespace CatShop.Models
{
    using System.ComponentModel.DataAnnotations;

    public class Cat
    {

        public Cat() { }

        public Cat(string name, string nickname, double price)
        {
            this.Name = name;
            this.Nickname = nickname;
            this.Price = price;
        }

        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Nickname { get; set; }

        [Required]
        public double Price { get; set; }
    }
}
